### PR TITLE
Add support for RSA PKCSv1.5 signatures using RIPEMD-160

### PR DESCRIPTION
**Pull request by jack-fortanix**
Discussed in #2620

## Status
Ready for review

## Requires Backporting
New feature, so I think backporting not necessary.

## Todos
- [x] Tests
- [ ] Documentation - I was not sure what if anything needs to be updated for this.
- [x] Changelog updated

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -7,6 +7,8 @@ Features
      rfc 5280 section 4.2.1.4.
    * It is now possible to use NIST key wrap mode via the mbedtls_cipher API.
      Contributed by Jack Lloyd and Fortanix Inc.
+   * It is now possible to perform RSA PKCS v1.5 signatures with RIPEMD-160 digest.
```
Comment (RonEld) [inline ChangeLog]: Minor: should be at the end of the list

**Comment by jack-fortanix** [comment]
@RonEld Thanks for reviewing. I added a test of the OID->MD function in c8fc019

**Comment by jack-fortanix** [comment]
@RonEld I added a OID->MD test but it is failing in different configurations (eg when SHA512 or RIPEMD is disabled), because the OIDs in `oid_md_alg` table are only defined if the cooresponding hash function is enabled in the build. This is unlike the situation for the other OIDs tested in that test (like X.509 extensions) which are always enabled regardless of how the library is configured. Any hints on how to address this?

IMO the correct behavior is to simply always define all of the MD OIDs, regardless of if they are enabled. After all, the OID 2.16.840.1.101.3.4.2.3 is always the identifier for SHA-512, even if a particular build of mbedtls doesn't support SHA-512. But maybe allowing this function to return success in this situation has follow-on effects within the library that I don't realize.

Another alternative would be to (within the test) always accept MBEDTLS_ERR_OID_NOT_FOUND as a valid response from mbedtls_oid_get_md_alg, and only check it against the expected result if the return value is 0.

File: tests/suites/test_suite_oid.function [hunk 2]
```diff
@@ -74,7 +76,34 @@ void oid_get_x509_extension( data_t *oid, int exp_type )
...
+    int ret;
+    mbedtls_md_type_t md_id = 0;
+
+    md_oid.tag = MBEDTLS_ASN1_OID;
+    md_oid.p = oid->x;
+    md_oid.len = oid->len;
+
+    ret = mbedtls_oid_get_md_alg( &md_oid, &md_id );
+
+    if( exp_md_id == 0 )
```
[the remark this answers is not in the document]
Comment (jack-fortanix) [reply tests/suites/test_suite_oid.function]: Thanks for comments and also pointing me to the test script, I didn't know about that.
I am not quite sure what to make of this one suggestion, can I do this without having to edit the overall test framework?

I did change the expected result to -1 for the invalid oid test.

**Comment by jack-fortanix** [comment]
@RonEld Friendly ping, I think this just needs a last look from you

**Pull request merged**